sharedbuffers/mapped_struct.py

```diff
@@ -2825,7 +3301,10 @@ def build(cls, initializer, destfile = None, tempdir = None):
         try:
             dtypemax = ~dtype(0)
         except:
-            dtypemax = ~0
+            try:
```

Comment:
```
>>> import numpy
>>> a = numpy.array([1])
>>> a.dtype
dtype('int64')
>>> dtype = a.dtype
>>> dtype(0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'numpy.dtype' object is not callable
>>> 
```

It depends on where the dtype comes from. When using the ones at module-level, it works:

```
>>> dtype = numpy.uint64
>>> ~dtype(0)
18446744073709551615
```

So that except is just in case we get another dtype (the attribute can be overridden by subclasses)